src/test/kotlin/no/nav/syfo/cronjob/virksomhetsnavn/PersonOppfolgingstilfelleVirksomhetsnavnCronjobSpek.kt

```diff
@@ -0,0 +1,338 @@
+                    oversikthendelseType = OversikthendelseType.OPPFOLGINGSPLANLPS_BISTAND_MOTTATT,
+                    personIdent = personIdentDefault.value,
+                )
+                oversiktHendelseService.oppdaterPersonMedHendelse(
+                    oversiktHendelse = oversiktHendelseOPLPSBistandMottatt,
+                )
+
+                kafkaOppfolgingstilfellePersonService.pollAndProcessRecords(
+                    kafkaConsumerOppfolgingstilfellePerson = mockKafkaConsumerOppfolgingstilfellePerson,
+                )
+
+                val recordValue = kafkaOppfolgingstilfellePersonServiceRecordRelevant.value()
+
+                val pPersonOversiktStatusList = database.connection.use { connection ->
+                    connection.getPersonOversiktStatusList(
+                        fnr = recordValue.personIdentNumber,
+                    )
+                }
+
+                pPersonOversiktStatusList.size shouldBeEqualTo 1
+
+                val pPersonOversiktStatus = pPersonOversiktStatusList.first()
+
+                pPersonOversiktStatus.fnr shouldBeEqualTo recordValue.personIdentNumber
+                pPersonOversiktStatus.enhet shouldBeEqualTo oversiktHendelseOPLPSBistandMottatt.enhetId
+                pPersonOversiktStatus.veilederIdent.shouldBeNull()
+
+                pPersonOversiktStatus.motebehovUbehandlet.shouldBeNull()
+                pPersonOversiktStatus.moteplanleggerUbehandlet.shouldBeNull()
+                pPersonOversiktStatus.oppfolgingsplanLPSBistandUbehandlet shouldBeEqualTo true
+
+                checkPPersonOversiktStatusOppfolgingstilfelle(
+                    pPersonOversiktStatus = pPersonOversiktStatus,
+                    kafkaOppfolgingstilfellePerson = recordValue,
+                )
+
+                database.connection.use { connection ->
+                    val pPersonOppfolgingstilfelleVirksomhetList =
+                        connection.getPersonOppfolgingstilfelleVirksomhetList(
+                            pPersonOversikStatusId = pPersonOversiktStatus.id,
+                        )
+
+                    checkPPersonOppfolgingstilfelleVirksomhet(
+                        pPersonOppfolgingstilfelleVirksomhetList = pPersonOppfolgingstilfelleVirksomhetList,
+                        kafkaOppfolgingstilfellePerson = recordValue,
+                        updated = false,
+                    )
+                }
+
+                runBlocking {
+                    val result = personOppfolgingstilfelleVirksomhetnavnCronjob.runJob()
+
+                    result.failed shouldBeEqualTo 1
+                    result.updated shouldBeEqualTo 1
+                }
+
+                runBlocking {
+                    val result = personOppfolgingstilfelleVirksomhetnavnCronjob.runJob()
+
+                    result.failed shouldBeEqualTo 1
+                    result.updated shouldBeEqualTo 0
+                }
```

Comment:
Jeg skjønner ikke hvorfor man får 1 feil og 1 oppdatering første runde, og 1 feil andre gang 🤷

Reply:
Hovedtanken er er å sjekke det kun gjøres en oppdatering, altså oppdatering maksimalt  1 gang uavhengig av antall kjøringer.
I tillegg greit å sjekke at det feiler og det feiler for denne:
```
   get("$EREG_PATH/${VIRKSOMHETSNUMMER_NO_VIRKSOMHETSNAVN.value}") {
                call.respond(HttpStatusCode.InternalServerError, "")
            }
```
Men ser at hensikten er vanskelig å lese av koden og at det er ganske implisitt. Kan se om jeg kan gjøre noe som gjør det lettere å følge.

Reply:
Nå endret jeg det til å være 1 test som kun test om den får oppdatert og la til en ny test som tester at oppdatering feilet.